### DISCUSSION TITLE
autogluon in kaggle notebook

### DISCUSSION THREAD

**Discussion by eefalco** [community/community#46333] [Other Feature Feedback, Questions, & Ideas] [answered]
I am trying to use the autogluon timeseries predictor in a kaggle notebook. I copied the code from
https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-quickstart.html
into a kaggle notebook. When I try to run this part:
---
predictor = TimeSeriesPredictor(
    prediction_length=48,
    path="autogluon-m4-hourly",
    target="target",
    eval_metric="sMAPE",
)

predictor.fit(
    train_data,
    presets="medium_quality",
    time_limit=600,
    hyperparameters=hypers,
)
----
It fails with an inscrutable error:
----
TypeError                                 Traceback (most recent call last)
/tmp/ipykernel_27/60795051.py in <module>
     12     presets="medium_quality",
     13     time_limit=600,
---> 14     hyperparameters=hypers,
     15 )
..... lots of unhelpful text
TypeError: n needs to be an int or a list/set/tuple of ints
----
It still happens if I drop hyperparameters, time_limit, etc.
There is no n that I can see. Anyway, if anyone has an idea, please 
let me know.

**Comment by shayan1234554321** [comment, marked as answer]
The error message "TypeError: n needs to be an int or a list/set/tuple of ints" suggests that one of the arguments being passed to the fit method is not of the expected type ⚠️ . The specific argument that is causing the issue is not clear from the information provided, but it is possible that hypers is not of type int, list, set, or tuple.

To resolve this issue ✅ , you will need to examine the value of hypers and ensure that it is of the correct type. If it is a list, set, or tuple, make sure that all elements are integers ✍🏼 . If it is an integer, check that it is a valid value.

If you are still encountering this issue after verifying the type of hypers, you can try printing the value of hypers just before the fit method is called to see if there is any additional information that can help you identify the cause of the error.

👉 Additionally, you may want to refer to the documentation for the fit method or the TimeSeriesPredictor class to ensure that you are using it correctly and that the arguments you are passing are valid.

**Reply by eefalco** [reply]
Thanks for the reply. It turns out that even if I remove  every optional parameter from my call to predictor, I still get the same TypeError. I also discovered that the error is in a call to groupby and nth. I think only a developer type could get to the root cause. It should not be difficult to replicate the error. 
However, I found that if I use autogluon 0.5.2 instead of the newer 0.6.2, it works fine with all the parameters I used. 
It looks like a bug was introduced in the new version of autogluon.

**Reply by shayan1234554321** [reply]
Great!! That you found out, please mark the answer as an answer, may it help others ✅

**Reply by jiheek** [reply]
@shayan1234554321  I'm also getting the same error even if I use autogluon 0.5.2
was there anything found as a solution to this error? I think it is an internal bug somewhere

**Reply by jiheek** [reply]
when I run `predictor = TimeSeriesPredictor(prediction_length=14).fit(train_data)`
following the example here : https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-indepth.html

I get this error :
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-167-5447678328e9> in <module>
----> 1 predictor = TimeSeriesPredictor(prediction_length=14).fit(train_data)
      2 predictor

/opt/conda/lib/python3.7/site-packages/autogluon/core/utils/decorators.py in _call(*args, **kwargs)
     28         def _call(*args, **kwargs):
     29             gargs, gkwargs = g(*other_args, *args, **kwargs)
---> 30             return f(*gargs, **gkwargs)
     31         return _call
     32     return _unpack_inner

/opt/conda/lib/python3.7/site-packages/autogluon/timeseries/predictor.py in fit(self, train_data, tuning_data, time_limit, presets, hyperparameters, hyperparameter_tune_kwargs, enable_ensemble, random_seed, **kwargs)
    494         model_types = self._trainer.get_models_attribute_dict(attribute="type")
    495         model_typenames = {key: model_types[key].__name__ for key in model_types}
--> 496         unique_model_types = set(model_typenames.values())  # no more class info
    497 
    498         # all fit() information that is returned:

/opt/conda/lib/python3.7/site-packages/autogluon/timeseries/learner.py in fit(self, train_data, val_data, hyperparameters, hyperparameter_tune_kwargs, **kwargs)
     72         hyperparameter_tune_kwargs: Optional[Union[str, dict]] = None,
     73         time_limit: Optional[int] = None,
---> 74         **kwargs,
     75     ) -> None:
     76         self._time_limit = time_limit

/opt/conda/lib/python3.7/site-packages/autogluon/timeseries/learner.py in _fit(self, train_data, val_data, hyperparameters, hyperparameter_tune_kwargs, time_limit, **kwargs)
    112         self,
    113         data: TimeSeriesDataFrame,
--> 114         model: Optional[AbstractTimeSeriesModel] = None,
    115         **kwargs,
    116     ) -> TimeSeriesDataFrame:

/opt/conda/lib/python3.7/site-packages/autogluon/timeseries/splitter.py in split(self, ts_dataframe, prediction_length)

/opt/conda/lib/python3.7/site-packages/autogluon/timeseries/splitter.py in _split(self, ts_dataframe, prediction_length)

/opt/conda/lib/python3.7/site-packages/autogluon/timeseries/dataset/ts_dataframe.py in slice_by_timestep(self, start_index, end_index)

/opt/conda/lib/python3.7/site-packages/pandas/core/groupby/groupby.py in nth(self, n, dropna)
   2302         valid_containers = (set, list, tuple)
   2303         if not isinstance(n, (valid_containers, int)):
-> 2304             raise TypeError("n needs to be an int or a list/set/tuple of ints")
   2305 
   2306         if not dropna:

TypeError: n needs to be an int or a list/set/tuple of ints

**Reply by shayan1234554321** [reply]
The error message indicates that the TimeSeriesPredictor class is trying to slice the train_data using the split method of the TimeSeriesSplitter class, and there is an issue with the input arguments. 😶

The error message specifically says that the argument n in the nth method of the Pandas GroupBy object needs to be an integer 🥴 or a list/set/tuple of integers. This suggests that the input data train_data may not be in the correct format.

1. Make sure that train_data is a TimeSeriesDataFrame object that has been properly formatted and prepared for training.
2. Check the documentation for the TimeSeriesPredictor class and make sure that you are passing the correct arguments to the fit method.
3. Try passing a different value for prediction_length to see if the error persists. 🫢

**Reply by aadilmalik994** [reply]
I installed below dependencies and it worked for me:
statsmodels
sktime
pmdarima
sktime[all_extras]
networkx
mxnet==1.9
autogluon.timeseries==0.5.2
autogluon.tabular

**Comment by AmirZidi** [comment]
I faced the same issue and it turned out to be a problem of compatibility between autogluon and pandas. If you are using atuogluon >0.6.x make sure you have pandas > 1.4.x.

**Reply by RichardYangHe** [reply]
Thank you! This sloved my problem. 
Pip install pandas==01.5
